docs/components/SegmentedControl/SegmentedControl.stories.mdx

```diff
@@ -0,0 +1,111 @@
+
+## Design & usage guidelines
+
+SegmentedControl should only use up to five options with succinct labels. The
+component will also grow quite wide with more options and can become unusable on
+smaller screens. In that case, it is advised to put the SegmentedControl in a
+parent container with a fixed width. Be sure to avoid truncating the labels.
+
+## Content guidelines
+
+Concise [Text](../?path=/docs/components-text-and-typography-text--docs) should
+be used as the primary format for option labels.
+
+<Canvas>
+  <SegmentedControl defaultOption="option1">
+    <SegmentedControl.Option value="option1">Option 1</SegmentedControl.Option>
+    <SegmentedControl.Option value="option2">Option 2</SegmentedControl.Option>
+    <SegmentedControl.Option value="option3">Option 3</SegmentedControl.Option>
+    <SegmentedControl.Option value="option4">Option 4</SegmentedControl.Option>
+    <SegmentedControl.Option value="option5">Option 5</SegmentedControl.Option>
+  </SegmentedControl>
+</Canvas>
+
+Using [Icons](../?path=/docs/components-images-and-icons-icon--docs) as labels
+should only be used if the icon has been tested as understandable in isolation.
+
+<Canvas>
+  <SegmentedControl defaultOption="calendar">
+    <SegmentedControl.Option value="calendar">
+      <Icon name="calendar" />
+    </SegmentedControl.Option>
+    <SegmentedControl.Option value="phone">
+      <Icon name="phone" />
+    </SegmentedControl.Option>
+    <SegmentedControl.Option value="availability">
+      <Icon name="availability" />
+    </SegmentedControl.Option>
+  </SegmentedControl>
+</Canvas>
+
+Mixing content types within a single label, like using an icon with text, should
+be avoided. This would be an inefficient use of the SegmentedControl's limited
+horizontal space.
+
+Mixing content types across options should also be avoided. This would create an
+inconsistent experience for users and lead to a jarring visual rhythm.
+
+If you do not have enough horizontal room to allow for readable labels, consider
+using [Menu](/?path=/docs/components-navigation-menu--docs) as an alternative
+single-selection method for modifying the view. Otherwise, truncation will
+prevent the user from reading the label and a label with wrapped text will
+create layout inconsistencies.
+
+## Controlled and Uncontrolled
+
+Use a
+[controlled](../?path=/story/components-selections-segmentedcontrol-web--basic-controlled)
+SegmentedControl when you need to manage the state of the selected option from a
+parent component. This is useful when the selected option needs to be
+synchronized with other state or when you need to perform side effects when the
+selected option changes.
+
+Use an
+[uncontrolled](../?path=/story/components-selections-segmentedcontrol-web--basic-uncontrolled)
+version when the SegmentedControl can manage it's own state. This is useful when
+the selected option does not need to be synchronized with other state or when
+you need a simple implementation.
```

Comment:
This is a good explanation of when to use a controlled vs uncontrolled segmented control. It doesn't really explain how to actually make it controlled vs uncontrolled. If I was unfamiliar with the concept I don't think I would know how to go about it. Also, looking at `SegmentedControlProvider` `isControlled` is set to true if both `selectedValue` and `onSelectedValue` are set. So what happens if I only set one? If this leads to weird or unexpected behaviour I think it should be documented. 

Sometimes we can rely on the prop descriptions but looking here:
![Screenshot 2024-11-13 at 11 09 20 AM](https://github.com/user-attachments/assets/44ec6698-7b76-4d7d-a34b-efeeb3cbb9a9)
If I didn't know anything about the component I wouldn't know how these three props connect to controlled vs uncontrolled

Reply:
Great points!
I added more info on the Docs page about how to implement controlled vs uncontrolled. I created code examples and described what props to use for each variation. Updated here 2caca5d47ba6fec9e469e27187d5f5e5e97d48bf
I hear what you're saying about explaining the unexpected behaviour if you deviate from that, however I prefer to demonstrate/explain the happy path, rather than add more documentation on what will go wrong if you don't follow the documentation. If you feel strongly about that lets chat. I personally feel like it may confuse the reader but I'm open for more conversation on that!

The prop descriptions were leaving a lot to be desired 😆  I added more detail here 39d330bc2bf949ff8cb99a2a404044c46f640cfb